contracts/core/fee-calculators/PerformanceFeeCalculator.sol

```diff
@@ -0,0 +1,383 @@
+        uint256 _maximumProfitFeePercentage,
+        uint256 _maximumStreamingFeePercentage
+    )
+        public
+    {
+        core = _core;
+        oracleWhiteList = _oracleWhiteList;
+        maximumProfitFeePercentage = _maximumProfitFeePercentage;
+        maximumStreamingFeePercentage = _maximumStreamingFeePercentage;
+    }
+
+    /* ============ External Functions ============ */
+
+    /*
+     * Called by RebalancingSetToken, parses bytedata then assigns to correct FeeState struct. 
+     *
+     * @param  _feeCalculatorData       Bytestring encoding fee parameters for RebalancingSetToken
+     */
+    function initialize(
+        bytes calldata _feeCalculatorData
+    )
+        external
+    {
+        // Parse fee data into struct
+        InitFeeParameters memory parameters = parsePerformanceFeeCallData(_feeCalculatorData);
+
+        // Validate fee data
+        validateFeeParameters(parameters);
+
+        // Set fee state for new caller
+        feeState[msg.sender].profitFeeFrequency = parameters.profitFeeFrequency;
+        feeState[msg.sender].highWatermarkResetFrequency = parameters.highWatermarkResetFrequency;
+        feeState[msg.sender].profitFeePercentage = parameters.profitFeePercentage;
+        feeState[msg.sender].streamingFeePercentage = parameters.streamingFeePercentage;
+        feeState[msg.sender].lastProfitFeeTimestamp = block.timestamp;
+        feeState[msg.sender].lastStreamingFeeTimestamp = block.timestamp;
+        feeState[msg.sender].highWatermark = calculateRebalancingSetValue(msg.sender);
+    }
+
+    /*
+     * Calculates total inflation percentage in order to accrue fees to manager. Profit fee calculations
+     * are net of streaming fees, so streaming fees are applied first then profit fees are calculated.  
+     *
+     * @param  uint256       Percent inflation of supply
+     */
+    function getFee()
+        external
+        view
+        returns (uint256)
+    {
+        uint256 streamingFee = calculateStreamingFee();
+
+        // If time since last profit fee exceeds profitFeeFrequency then calculate profit fee else 0.
+        uint256 profitFee = 0;
+        if (exceedsProfitFeeFrequency()) {
+            profitFee = calculateProfitFee(streamingFee);
+        }
+
+        return streamingFee.add(profitFee);
+    }
+
+    /*
+     * Calculates total inflation percentage in order to accrue fees to manager. Profit fee calculations
+     * are net of streaming fees, so streaming fees are applied first then profit fees are calculated. 
+     * Additionally, fee state is set timestamps are updated for each fee type and the high watermark is
+     * reset if time since last profit fee exceeds the highWatermarkResetFrequency. 
+     *
+     * @param  uint256       Percent inflation of supply
+     */
+    function updateAndGetFee()
+        external
+        returns (uint256)
+    {
+        uint256 streamingFee = calculateStreamingFee();
+
+        uint256 profitFee = 0;
```

Comment:
Could deduplicate this code by creating an internal function called `getProfitFee` - which returns the value based on this logic here.